src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4947,7 +4945,7 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
   const QString cacheLayer { QStringLiteral( "ovrlaylyr:%1" ).arg( cacheBase ) };
   const QString cacheIndex { QStringLiteral( "ovrlayidx:%1" ).arg( cacheBase ) };
 
-  if ( !context->hasCachedValue( cacheLayer ) )
+  if ( !context->hasCachedValue( cacheLayer ) ) // should check for same crs. if not the same we could think to reproject target layer before charging cache
```

Comment:
Good thinking! Added this to the todo list (but it should be done in the call to `materialize`, once cached we should just be able to assume that CRSes match